packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -250,22 +250,47 @@ export const internals = Joi.extend([
 
     base: Joi.object(),
     coerce(value: any, state: State, options: ValidationOptions) {
-      // If value isn't defined, let Joi handle default value if it's defined.
-      if (value !== undefined && !isPlainObject(value)) {
-        return this.createError('object.base', { value }, state, options);
+      if (value === undefined || isPlainObject(value)) {
+        return value;
       }
 
-      return value;
+      if (options.convert && typeof value === 'string') {
+        try {
+          const parsed = JSON.parse(value);
+          if (isPlainObject(parsed)) {
+            return parsed;
+          }
+          return this.createError('object.base', { value: parsed }, state, options);
```

Comment:
I'm using the parsed value to create the error message here. `"[1,2,3]"` will fails with an error message stating that it received an array instead of an object, even if the actual raw input was a string.I think using the parsed value type makes more sense / is more clear, as least for route validation where the parsing is implicit for `query`  and or `multipart/form-data` data.